### PR TITLE
fix issue-7656: When using multiple protocols and having port specifi…

### PR DESCRIPTION
**Pull request by changfubai**
…ed, service can not be started.

## What is the purpose of the change

fix #7656: When using multiple protocols and having port specified, service can not be started.

submit it into 3.0.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java [hunk 1]
```diff
@@ -756,17 +756,17 @@ private Integer findConfigedPorts(ProtocolConfig protocolConfig,
...
         Integer portToRegistry = parsePort(portToRegistryStr);
-        if (portToRegistry == null) {
-            portToRegistry = portToBind;
+        if (portToRegistry != null) {
+            portToBind = portToRegistry;
         }
 
-        return portToRegistry;
+        // save bind port, used as url's key later
+        map.put(BIND_PORT_KEY, String.valueOf(portToBind));
```
Comment (changfubai) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java:767]: if registryPort is not null,  make bindPort = registryPort
put bindPort into map.
return bindPort.

**Comment by changfubai** [comment]
太难了，跑了两次单测都不过

**Review by wangchengming666 — APPROVED** [review]
LGTM

**Comment by AlbumenJ** [comment]
Please resolve conflicts

**Comment by changfubai** [comment]
> Please resolve conflicts

I resolved conflicts, help me review please!

**Pull request merged**